### PR TITLE
v1.9 backports 2020-11-09

### PR DESCRIPTION
**Pull request by jrajahalme**
* #13890 -- maglev: Convert permutation 2D array to 1D (@brb)
 * #13908 -- bpf: don't access TCP flags for non initial IPv4 fragments (@jibi)
 * #13922 -- bpf: Compile test POLICY_VERDICT_NOTIFY (@pchaigno)
 * #13942 -- doc: Fix a typo (@aditighag)
 * #13878 -- gke: add resource quotas for Cilium Namespace (@aanm)
 * #12658 -- test: Move RuntimeVerifier to K8sVerifier (@pchaigno)
 * #13906 -- docs: ipam - update Azure Private IP link (@ti-mo)
 * #13937 -- change default docker image repository from docker.io to quay.io (@aanm)
 * #13916 -- ipam/allocator: Fix nil check on node CIDR (@christarazi)
 * #13949 -- install/kubernetes: add check-docker-images target to phony targets (@tklauser)
 * #13944 -- test: Increase timeout on privileged unit tests (@pchaigno)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 13890 13908 13922 13942 13878 12658 13906 13937 13916 13949 13944; do contrib/backporting/set-labels.py $pr done 1.9; done
```

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-backport-1.9

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
```
11:38:19  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

Apparently some piece of CI is still dependent on dockerhub :(

**Comment by jrajahalme** [comment]
```$ test/provision/container-images.sh test_images `pwd`
Downloading all container images needed for tests
not cached in VM: 	docker.io/byrnedo/alpine-curl:0.1.8
pulling: 		docker.io/byrnedo/alpine-curl:0.1.8
not cached in VM: 	docker.io/cilium/demo-httpd:latest
pulling: 		docker.io/cilium/demo-httpd:latest
not cached in VM: 	docker.io/cilium/dnssec-client:v0.2
pulling: 		docker.io/cilium/dnssec-client:v0.2
not cached in VM: 	docker.io/cilium/docker-bind:v0.3
pulling: 		docker.io/cilium/docker-bind:v0.3
not cached in VM: 	docker.io/cilium/istio_pilot:1.5.9
pulling: 		docker.io/cilium/istio_pilot:1.5.9
not cached in VM: 	docker.io/cilium/istio_proxy:1.5.9
pulling: 		docker.io/cilium/istio_proxy:1.5.9
not cached in VM: 	docker.io/cilium/kafkaclient2:latest
pulling: 		docker.io/cilium/kafkaclient2:latest
not cached in VM: 	docker.io/cilium/python-bmemcached:v0.0.2
pulling: 		docker.io/cilium/python-bmemcached:v0.0.2
not cached in VM: 	docker.io/digitalwonderland/zookeeper:latest
pulling: 		docker.io/digitalwonderland/zookeeper:latest
not cached in VM: 	docker.io/library/alpine:3.9
pulling: 		docker.io/library/alpine:3.9
not cached in VM: 	docker.io/library/busybox:1.31.1
pulling: 		docker.io/library/busybox:1.31.1
not cached in VM: 	docker.io/library/cassandra:3.11.3
pulling: 		docker.io/library/cassandra:3.11.3
not cached in VM: 	docker.io/library/memcached:1.6.6-alpine
pulling: 		docker.io/library/memcached:1.6.6-alpine
not cached in VM: 	docker.io/library/redis:6.0.5
pulling: 		docker.io/library/redis:6.0.5
not cached in VM: 	docker.io/tgraf/netperf:v1.0
pulling: 		docker.io/tgraf/netperf:v1.0
not cached in VM: 	docker.io/wurstmeister/kafka:2.11-0.11.0.3
pulling: 		docker.io/wurstmeister/kafka:2.11-0.11.0.3
not cached in VM: 	quay.io/coreos/etcd:v3.4.7
pulling: 		quay.io/coreos/etcd:v3.4.7

File: test/k8sT/manifests/test-verifier.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-verifier
+spec:
+  containers:
+  - name: cilium-builder
+    image: quay.io/cilium/cilium-builder:2020-10-20
```
Comment (pchaigno) [inline test/k8sT/manifests/test-verifier.yaml]: Maybe we should switch this to use `quay.io/cilium/cilium-builder:2020-11-09-v1.9@sha256:401d3bc1464c1daf54f4205aea936cb3fb90f0ae00c03e2146ebf2a9d23d5651`, i.e., the tag we use in the v1.9 branch?

**Comment by joestringer** [comment]
net-next and GKE builds failed due to dockerhub pull ratelimiting. I will re-kick in the hope they are rescheduled to nodes that are not subject to ratelimiting.

**Comment by joestringer** [comment]
retest-net-next

**Comment by joestringer** [comment]
retest-gke

**Comment by joestringer** [comment]
retest-gke

**Review by christarazi — APPROVED** [review]
LGTM for my commits

**Comment by joestringer** [comment]
Green runs except for GKE which won't run due to dockerhub ratelimits. Will fix up the feedback from @pchaigno and merge.

**Pull request merged**